### PR TITLE
fix: [CDS-87146]: Code module old nav to new nav redirects

### PR DESCRIPTION
**Pull request by avinashmadhwani02**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

**Comment by avinashmadhwani02** [comment]
trigger messagecheck

**Review by MondalSayantan — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Comment by tan-nhu** [comment]
I tested with a couple of URLs, they are not redirected correctly.

OLD: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/code/default/CODE
REDIRECT TO: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/module/code/orgs/default/settings
SHOULD BE: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/module/code/orgs/default/projects/CODE

OLD: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/code/default/CODE/gitness
REDIRECT TO: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/module/code/orgs/default/projects/CODEgitness/repos
SHOULD BE: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/module/code/orgs/default/projects/CODE/repos/gitness

OLD: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/code/default/CODE/gitness/files/main/~/.githooks/pre-commit
REDIRECT TO: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/module/code/orgs/default/projects/CODEgitness/files/main/~/.githooks/pre-commit
SHOULD BE: https://localhost:8181/ng/account/_OKE5H2PQKOUfzFFDuD4FA/module/code/orgs/default/projects/CODE/repos/gitness/files/main/~/.githooks/pre-commit

Etc...

**Review by tan-nhu — CHANGES_REQUESTED** [review]
Tested with a bunch of old URLs, none worked with the redirect.

File: src/modules/RouteDestinationsV2.tsx [hunk 1]
```diff
@@ -59,6 +59,36 @@ export const RedirectToMode = ({ mode }: { mode?: NAV_MODE }): React.ReactElemen
...
+    projectIdentifier,
+    orgIdentifier
+  } = useParams<ModulePathParams & ProjectPathProps & { path: string }>()
+
+  const { currentMode } = useAppStore()
+
+  const finalMode = currentMode && isNavMode(currentMode) ? currentMode : module ? NAV_MODE.MODULE : NAV_MODE.ADMIN
+
+  const locationParams = useLocation()
+  const path = !_path ? 'repos' : _path !== 'search' ? `repos/${_path}` : _path
```
Comment (tan-nhu) [inline src/modules/RouteDestinationsV2.tsx:75]: This constructs correct CODE URLs for V2.

**Review by tan-nhu — APPROVED** [review]

**Review by hitesharinga — APPROVED** [review]

**Comment by cjlee01** [comment]
Lgtm tested some of the cases as well

**Pull request merged**